tests/conftest.py

```diff
@@ -19,3 +19,19 @@ def teardown():
     request.addfinalizer(teardown)
 
     return flaskr.app
+
+
+@pytest.fixture(scope='function')
+def base_url(live_server):
```

Comment:
From pytest-flask, it's a fixture that starts an in-process copy of the application. We do this for every test.